theano/gpuarray/tests/test_linalg.py

```diff
@@ -293,3 +302,131 @@ def test_gpu_singular_values(self):
 
         A_val = rand(100, 50).astype('float32')
         utt.assert_allclose(f_cpu(A_val), f_gpu(A_val))
+
+    def run_gpu_cholesky(self, A_val, lower=True):
+        A = theano.tensor.fmatrix("A")
+        f = theano.function([A], GpuMagmaCholesky(lower=lower)(A),
+                            mode=mode_with_gpu.excluding('cusolver'))
+        return f(A_val)
+
+    def check_cholesky(self, N, lower=True, rtol=None, atol=None):
+        A = rand(N, N).astype('float32')
+        # ensure that eigenvalues are not too small which sometimes results in
+        # magma failure due to gpu limited numerical precision
+        D, W = np.linalg.eigh(A)
+        D[D < 1] = 1
+        V_m = np.zeros_like(A)
+        np.fill_diagonal(V_m, D)
+        A = np.dot(np.dot(W.T, V_m), W)
+        L = self.run_gpu_cholesky(A, lower=lower)
+        if not lower:
+            L = L.T
+        utt.assert_allclose(np.dot(L, L.T), A, rtol=rtol, atol=atol)
+
+    def test_gpu_cholesky(self):
+        self.check_cholesky(1000, atol=1e-3)
+        self.check_cholesky(1000, lower=False, atol=1e-3)
+
+    def test_gpu_cholesky_opt(self):
+        A = theano.tensor.matrix("A", dtype="float32")
+        fn = theano.function([A], cholesky(A), mode=mode_with_gpu.excluding('cusolver'))
+        assert any([isinstance(node.op, GpuMagmaCholesky)
+                    for node in fn.maker.fgraph.toposort()])
+
+    def test_gpu_cholesky_inplace(self):
+        N = 1000
+        A = rand(N, N).astype('float32')
+        A = np.dot(A.T, A)
```

Comment:
On my computer, tests still fail at this point (`test_gpu_cholesky_inplace()`). When I replaced this line with the same changes you added into check_cholesky(), then all tests pass. So, I think you should update computation of A here as you do i check_cholesky().